adapters/yieldlab/yieldlab.go

```diff
@@ -212,7 +212,15 @@ func (a *YieldlabAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externa
 		Bids:     []*adapters.TypedBid{},
 	}
 
-	for i, bid := range bids {
+	adslotToImpMap := make(map[string]*openrtb2.Imp)
+	for i := 0; i < len(internalRequest.Imp); i++ {
+		adslotID := a.extractAdslotID(internalRequest.Imp[i])
```

Comment:
It is possible that `extractAdslotID()` returns a nil value if there is an issue unmarshalling the bidder parameters. It also seems that your bidding depends on `adslotID` being unique to each imp in the request. But I don't see anywhere you enforce this in `MakeRequests()`. You probably want to throw out any imps with duplicate adslots in the `MakeRequests()` stage, and return either a warning or an error about the issue. A nil adslot will likely mean that the imp will get no. bids anyway, and if it does will just be skipped below so again not a fatal issue.

The possibility of duplicate adslots in Imps does seem like something that should be addressed, as it can otherwise lead to bids being associated with the wrong imp.

Reply:
Hi @hhhjort, thanks for your input!

You are correct that our bidding depends on adslotID being unique to each imp in the request and that we don't enforce this within Prebid. This is enforced from our servers though: When we get a bid request for adslot IDs 1, 1, 1 and 2, this will actualy be flattened to a request for adslot IDs 1 and 2 in our servers, and so only one bid per adslotID will be returned.

This is not specific to how we handle Prebid by the way, but also applies to any header bidding integration with Yieldlab and is something that is being pointed out to every publisher once they start working with us.

Does this ease your concerns?

Reply:
There is the issue that you map the bids back into the imps via the adslot ID So either when you flatten on the servers, you need to pick an ad that should be valid for all the relevant imps, or you run the danger of delivering a bid for an imp that that bid is not valid for. (perhaps a wrong size banner if the adslots are not coded for a specific size.) If the bid will be valid for all imps with the same adslot, then I don't have an issue.

Reply:
`If the bid will be valid for all imps with the same adslot, then I don't have an issue.`

Yes, that will be the case! Thanks for all your help!